schema/collection.js

```diff
@@ -0,0 +1,85 @@
+// @flow
+import type { GraphQLFieldConfig } from "graphql"
+import { pageable } from "relay-cursor-paging"
+import { connectionFromArraySlice } from "graphql-relay"
+import _ from "lodash"
+import gravity from "../lib/loaders/gravity"
+import cached from "./fields/cached"
+import { artworkConnection } from "./artwork"
+import { queriedForFieldsOtherThanBlacklisted, parseRelayOptions } from "../lib/helpers"
+import { GravityIDFields, NodeInterface } from "./object_identification"
+import { GraphQLObjectType, GraphQLString, GraphQLNonNull, GraphQLBoolean } from "graphql"
+
+// Note to developers working on collections, the staging server does not get a copy
+// of all artwork follows, so you will need to add some each week in order to have data
+// to work with.
+
+const CollectionType = new GraphQLObjectType({
+  name: "Collection",
+  interfaces: [NodeInterface],
+  isTypeOf: obj => _.has(obj, "name") && _.has(obj, "private") && _.has(obj, "default"),
+  fields: {
+    ...GravityIDFields,
+    cached,
+    artworks_connection: {
+      type: artworkConnection,
+      args: pageable({}),
+      resolve: ({ id }, options, request, { rootValue: { accessToken, userID } }) => {
+        const gravityOptions = parseRelayOptions(options)
+
+        gravityOptions.total_count = true
+        gravityOptions.user_id = userID
+
+        delete gravityOptions.page // this can't also be used with the offset in gravity
+        return gravity
+          .with(accessToken, { headers: true })(`collection/${id}/artworks`, gravityOptions)
+          .then(({ body, headers }) => {
+            return connectionFromArraySlice(body, options, {
+              arrayLength: headers["x-total-count"],
+              sliceStart: gravityOptions.offset,
+            })
+          })
+      },
+    },
+    description: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    default: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+    },
+    name: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+    private: {
+      type: new GraphQLNonNull(GraphQLBoolean),
+    },
+    slug: {
+      type: new GraphQLNonNull(GraphQLString),
+    },
+  },
+})
+
+const Collection: GraphQLFieldConfig<CollectionType, *> = {
+  type: CollectionType,
+  args: {
+    id: {
+      description: "The slug or ID of the Collection",
+      type: new GraphQLNonNull(GraphQLString),
+    },
+  },
+  resolve: (root, { id }, request, { fieldNodes, rootValue }) => {
+    // If you are just making an artworks call ( e.g. if paginating )
+    // do not make a Gravity call for the gene data.
+    const { accessToken, userID } = (rootValue: any)
```

Comment:
Cool I like this over specifying all of them in the method signature a few lines up (esp if there are more than 1). Otherwise it's a really long line length.